test/server.test.js

```diff
@@ -1,14 +1,536 @@
+          done();
+        });
+    });
+
+    it('should respond with 200 status and return an empty array when no products for that brand exist', (done) => {
+      const brandId = '6';
+
+      chai
+        .request(server)
+        .get(`/brands/${brandId}/products`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.should.be.empty;
+          done();
+        });
+    });
+
+    it('should get all the products of one specific brand when a valid brandId is supplied', (done) => {
+      const brandId = '2';
+
+      chai
+        .request(server)
+        .get(`/brands/${brandId}/products`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.length.should.be.eql(2);
+          done();
+        });
+    });
+  });
+});
+
+describe('Products', () => {
+  describe('/GET products', () => {
+    it('should get all the products when no search query is present', (done) => {
+      chai
+        .request(server)
+        .get('/products')
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.should.not.be.empty;
+          res.body[0].should.have.all.keys(
+            'id',
+            'categoryId',
+            'name',
+            'description',
+            'price',
+            'imageUrls',
+          );
+          done();
+        });
+    });
+
+    it('should treat an empty searchTerm the same as not supplying one', (done) => {
+      const searchTerm = '';
+
+      chai
+        .request(server)
+        .get(`/products?searchTerm=${searchTerm}`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.should.not.be.empty;
+          res.body[0].should.have.all.keys(
+            'id',
+            'categoryId',
+            'name',
+            'description',
+            'price',
+            'imageUrls',
+          );
+          done();
+        });
+    });
+
+    it('should respond with 200 and return an empty array if no products match search query', (done) => {
+      const searchTerm = 'blue';
+
+      chai
+        .request(server)
+        .get(`/products?searchTerm=${searchTerm}`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.should.be.empty;
+          done();
+        });
+    });
+
+    it('should return a filtered list of products based on search query otherwise', (done) => {
+      const searchTerm = 'glasses';
+
+      chai
+        .request(server)
+        .get(`/products?searchTerm=${searchTerm}`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.be.an('array');
+          res.body.should.not.be.empty;
+          res.body[0].should.have.all.keys(
```

Comment:
you are not really testing here the search functionality, you need to make sure the actual data is correct